docs/src/understanding_mooncake/algorithmic_differentiation.md

```diff
@@ -168,27 +168,32 @@ Following either resource will yield the derivative:
 ```math
 D f [X] (\dot{X}) = \dot{X}^\top X + X^\top \dot{X}
 ```
-Observe that this is indeed a linear operator (i.e. it is linear in its argument, ``\dot{X}``).
-(You can always plug it in to the definition of the Frechet derivative to confirm that it is indeed the derivative.)
+Observe that this is indeed a linear operator (i.e. it is linear in its argument, ``\dot{X}``). For the rest of this example, we use the matrix A as the coordinate-form representation of the operator ``Df\left[X\right]:\mathbb{R}^{N\times M}\rightarrow\mathbb{R}^{N\times M}``, with respect to the elementary basis of ``\mathbb{R}^{N\times M}``.
```

Comment:
Does the matrix `A` exist? Unless I'm mistaken, it's not the case that you can necessarily represent any linear operator on a matrix via multiplication with another matrix.

If you treat a matrix `X` of size `P x Q` as a vector of length `PQ`, you can certainly represent any linear transform via a matrix of size `PQ x PQ`, but (if my understanding of what you're doing here is correct), that's not what's happening here.

Reply:
Yes, you're right. I wrote things basis-free originally (via operators), then changed it hastily as I was typing for this example without realising I made this mistake.

I'll submit a change with a bit more discussion soon.

Reply:
Cool. FWIW, I think the basis-free approach that I originally adopted probably makes most sense here -- happy to be proven wrong though!

Reply:
Okay, I'm done writing the changes up. Let me figure out how to propose modifications to this PR.. sorry, I'm new at this.

By the way, it is possible to represent the finite-dimensional linear transformation Df evaluate at X as a Jacobian matrix. It is just quite scrambled up, so in this case it is indeed easier to work out something like a total derivative (what the docs call Frechet derivative) instead of explicitly computing the Jacobian.

For example, see the Jacobian matrix `J_p` in the following code. The variable `pf_p` is what you'd get with the analytical expression.
```
using LinearAlgebra, Random
import FiniteDiff

T = Float64
rng = Random.Xoshiro(0)

N = 2
M = 3

function eval_f(X)
    return X'*X
end

compute_J = xx->FiniteDiff.finite_difference_jacobian(eval_f, xx)

p = randn(rng, T, N, M)
J_p = compute_J(p)

compute_pushforward = (xx,vv)->(vv'*xx + xx'*vv)

v = randn(rng, T, N, M)
pf_p = compute_pushforward(p, v)
Jv_p = J_p*vec(v)

# The expression for the df_p can be expressed as the Jacobian.
pf_p_rec = reshape(Jv_p, size(pf_p))
println("Relative discrepancy between Jacobian vector product (matrix-form multiplied by 1-D array) vs. analytic expression for the differential of f")
@show norm(pf_p_rec - pf_p)/norm(pf_p)

println("The matrix-form of the differential, i.e., the Jacobian:")
display(J_p)
```

I get this in my REPL:
```
Discrepancy between Jacobian vector product (matrix-form multiplied by 1-D array) vs. analytic expression for the differential of f
norm(pf_p_rec - pf_p) / norm(pf_p) = 2.6244114944389197e-8
The matrix-form of the differential, i.e., the Jacobian:
9×6 Matrix{Float64}:
  1.88594  0.267846   0.0       0.0        0.0       0.0
  1.52507  0.123901   0.942971  0.133923   0.0       0.0
 -1.20577  0.311817   0.0       0.0        0.942971  0.133923
  1.52507  0.123901   0.942971  0.133923   0.0       0.0
  0.0      0.0        3.05014   0.247802   0.0       0.0
  0.0      0.0       -1.20577   0.311817   1.52507   0.123901
 -1.20577  0.311817   0.0       0.0        0.942971  0.133923
  0.0      0.0       -1.20577   0.311817   1.52507   0.123901
  0.0      0.0        0.0       0.0       -2.41154   0.623634
```